packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import app from 'state';
+import { useUpdateProfileByAddressMutation } from 'state/api/profiles';
+import { CWCheckbox } from 'views/components/component_kit/cw_checkbox';
+import { CWText } from 'views/components/component_kit/cw_text';
+import { CWButton } from 'views/components/component_kit/new_designs/CWButton';
+import { CWForm } from 'views/components/component_kit/new_designs/CWForm';
+import { CWTextInput } from 'views/components/component_kit/new_designs/CWTextInput';
+import { z } from 'zod';
+import './PersonalInformationStep.scss';
+import { personalInformationFormValidation } from './validations';
+
+type PersonalInformationStepProps = {
+  onComplete: () => void;
+};
+
+const PersonalInformationStep = ({
+  onComplete,
+}: PersonalInformationStepProps) => {
+  const { mutateAsync: updateProfile, isLoading: isUpdatingProfile } =
+    useUpdateProfileByAddressMutation();
+  const [emailBoundCheckboxKey, setEmailBoundCheckboxKey] = useState(1);
+
+  const handleSubmit = async (
+    values: z.infer<typeof personalInformationFormValidation>,
+  ) => {
+    await updateProfile({
+      address: app.user.activeAccount?.profile?.address,
+      chain: app.user.activeAccount?.profile?.chain,
+      name: values.username,
+      ...(values.email && {
+        email: values.email,
+      }),
+    });
+
+    // set email for notifications
+    if (values.email) {
+      app.user.updateEmail(values.email);
+    }
+
+    // TODO: update notification preferences here for
+    // values.enableAccountNotifications - does this mean all account notifications?
+    // values.enableProductUpdates - ?
+
+    onComplete();
+  };
+
+  const handleWatch = (
+    values: z.infer<typeof personalInformationFormValidation>,
+  ) => {
+    // if user enables an email bounded checkbox, we reset the checkbox
+    // when user clears the email field, as email is required for those
+    // bounded checkboxes
+    if (values.email.trim() === '') {
+      setEmailBoundCheckboxKey((key) => key + 1);
+    }
+  };
+
+  return (
+    <CWForm
+      className="PersonalInformationStep"
+      validationSchema={personalInformationFormValidation}
+      initialValues={{
+        enableAccountNotifications: false,
+        enableProductUpdates: false,
+      }}
+      onSubmit={handleSubmit}
+      onWatch={handleWatch}
+    >
+      {({ formState, watch }) => (
+        <>
+          <div className="username-section">
+            <CWTextInput
+              fullWidth
+              placeholder="Enter your user name"
+              label={
+                <CWText type="h4" fontWeight="semiBold">
+                  Enter a username&nbsp;<CWText type="b1">(required)</CWText>
+                </CWText>
+              }
+              name="username"
+              hookToForm
+            />
+            <CWButton
```

Comment:
shouldn't this button generate a username? it does nothing at the moment

Reply:
yes in https://github.com/hicommonwealth/commonwealth/issues/6751